source/_docs/automation.markdown

```diff
@@ -43,18 +43,51 @@ Actions are all about calling services. To explore the available services open t
 
 ### Automation initial state
 
-When you create a new automation, it will be enabled unless you explicitly add `initial_state: false` to it or turn it off manually via UI/another automation/developer tools. In case automations need to be always enabled or disabled when Home Assistant starts, then you can set the `initial_state` in your automations. Otherwise, the previous state will be restored.
+When you create a new automation, it will be enabled unless you explicitly add `initial_state: off` to it or turn it off manually via UI/another automation/developer tools. In case automations need to be always enabled or disabled when Home Assistant starts, then you can set the `initial_state` in your automations. Otherwise, the previous state will be restored.
 
 Please note that if for some reason Home Assistant cannot restore the previous state, it will result in the automation being enabled.
 
-```text
+```yaml
 automation:
 - alias: Automation Name
-  initial_state: false
+  initial_state: off
   trigger:
   ...
 ```
 
+### Automation Modes
+
+The automation's `mode` configuration option controls what happens when the automation is triggered while the actions are still running from a previous trigger. The default mode is `legacy`.
+
+Mode | Description
+-|-
+`legacy` | See [below](#legacy-mode).
+`error` | Raise an error. Previous run continues normally.
+`ignore` | Do not start a new run. Previous run continues normally.
+`parallel` | Start a new, independent run in parallel with previous runs which continue normally.
+`restart` | Start a new run after first stopping previous run.
+`queue` | Start a new run after all previous runs complete. Runs are guaranteed to execute in the order they were queued. The maximum number of queued up runs is controlled by the `queue_size` configuration option, which defaults to 10.
+
+#### Example Setting Automation Mode
+
+```yaml
+automation:
+- mode: queue
+  queue_size: 5
+  trigger:
+  ...
+```
+
+#### Legacy Mode
+
+<div class='note'>
+
+This mode is deprecated, and a warning to that effect will be issued at startup unless `mode: legacy` is specified.
```

Comment:
So how does our automation editor in the UI handle this?
Considering our users that don't have experience editing YAML...

Reply:
It doesn't, AFAIK. I don't use the Automation Editor, and I have no experience for working on the frontend. In the parent (code) PR @balloob said:

> btw, we should make sure that we update the automation and script editors. We should create an issue in the frontend for that.

& CC'd bramkragten

Reply:
Ok, but what is a user is fairly new to Home Assistant and never used YAML and just the frontend?

Reply:
Then like has always happened, they don't get to use the advanced feature until someone gets around to implementing it in the frontend. As I said, I can't help with that. And the default is legacy mode, which means it's still compatible with the frontend as-is in the interim.

Reply:
Sure but they will be confronted with an log messages and warnings, visible in the frontend, which was not their doing and have to learn YAML in order to fix it. I think we should have come up with a solution for the frontend before having merged this.

Reply:
Not my decision. Whatever y'all want to do is fine by me. Another option would be to automatically add `mode: legacy` to all automations & scripts, but I don't know how to do that either. I just tried to improve a script implementation that had serious flaws.

Reply:
@pnbruckner Don't get me wrong! You did an amazing job there m8. I've upgraded my production because of this 👍